go/vt/vtgate/router_select_test.go

```diff
@@ -505,6 +506,35 @@ func TestSelectIN(t *testing.T) {
 
 	sbc1.Queries = nil
 	sbc2.Queries = nil
+	_, err = routerExec(router, "select id from user where id in ::vals", map[string]interface{}{
+		"vals": []interface{}{int64(1), int64(3)},
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	wantQueries = []querytypes.BoundQuery{{
+		Sql: "select id from user where id in ::__vals",
+		BindVariables: map[string]interface{}{
+			"__vals": []interface{}{int64(1)},
+			"vals":   []interface{}{int64(1), int64(3)},
```

Comment:
A bind var could be reused in other parts of the query. It's unlikely. Maybe we can remove it for now, and then bring it back if someone runs into a genuine use case.